libs/utils/src/crashsafe.rs

```diff
@@ -44,7 +46,7 @@ pub fn create_dir_all(path: impl AsRef<Path>) -> io::Result<()> {
             None => {
                 return Err(io::Error::new(
                     io::ErrorKind::InvalidInput,
-                    format!("can't find parent of path '{}'", path.display()).as_str(),
+                    format!("can't find parent of path '{path}'",).as_str(),
```

Comment:
```suggestion
                    format!("can't find parent of path '{path}'"),
```

Seems this `as_str` removal is also unrelated, but it's not needed.